api/lib/resolvers.js

```diff
@@ -294,6 +300,41 @@ const transactionMetadata = async (
   }
 }
 
+const storeUser = async (_, { idToken, premium }) => {
+  try {
+    const decodedToken = await firebaseAdmin.auth().verifyIdToken(idToken)
+    // get user creation date
+    const userRecord = await firebaseAdmin.auth().getUser(decodedToken.uid)
+    // check if user already exists in DB
+    const storedUser = await database(config)('').getUser(decodedToken.uid)
+    // check if user already has premium as a custom claim
+    if (!userRecord.customClaims) {
+      // set premium field
+      await firebaseAdmin
+        .auth()
+        .setCustomUserClaims(decodedToken.uid, { premium: premium || false })
```

Comment:
what is the reason to use firebase claims?

Reply:
we need it to store the premium field inside the user. You put the docs reference in the issue description

Reply:
https://github.com/luniehq/lunie-api/issues/833